src/plone/restapi/services/aliases/add.py

```diff
@@ -84,9 +84,48 @@ class AliasesRootPost(Service):
     """Creates new aliases via controlpanel"""
 
     def reply(self):
-        data = json_body(self.request)
         storage = getUtility(IRedirectionStorage)
-        aliases = data.get("items", [])
+
+        if self.request.getHeader("Content-Type") == "text/csv":
+            content = self.request.get("BODY", "")
+            aliases = []
+            if not isinstance(content, bytes):
```

Comment:
Why not use the Python standard library [`csv`](https://docs.python.org/3/library/csv.html) throughout?

Reply:
I see, IMO its better to rely on `upload` method from the `RedirectControlPanel`. Instead of doing file validation at both places, So I've updated and used `upload` method.